x-pack/plugins/fleet/public/components/agent_enrollment_flyout/cloud_formation_instructions.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiButton, EuiSpacer, EuiCallOut } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+
+import { useGetSettings, useKibanaVersion, useStartServices } from '../../hooks';
+
+interface Props {
+  enrollmentAPIKey?: string;
+  cloudFormationTemplateUrl: string;
+}
+
+const createCloudFormationUrl = (
+  templateURL: string,
+  enrollmentToken: string,
+  fleetUrl: string,
+  kibanaVersion: string
+) => {
+  const cloudFormationUrl = templateURL
+    .replace('FLEET_ENROLLMENT_TOKEN', enrollmentToken)
+    .replace('FLEET_URL', fleetUrl)
+    .replace('KIBANA_VERSION', kibanaVersion);
+
+  return new URL(cloudFormationUrl).toString();
+};
+
+export const CloudFormationInstructions: React.FunctionComponent<Props> = ({
+  enrollmentAPIKey,
+  cloudFormationTemplateUrl,
+}) => {
+  const [fleetServer, setFleetServer] = useState<string | ''>();
+  const [isError, setIsError] = useState<boolean>(false);
+
+  const core = useStartServices();
+  const { data, isLoading } = useGetSettings();
+  const { notifications } = core;
+
+  const kibanaVersion = useKibanaVersion();
+
+  // Sets Fleet Server Host as the first fleet server host available from the settings
+  // Shows an error if no fleet server host is available
+  useEffect(() => {
+    if (isLoading) return;
+
+    const fleetServerHosts = data?.item.fleet_server_hosts;
+    if (fleetServerHosts !== undefined && fleetServerHosts.length !== 0) {
+      setFleetServer(fleetServerHosts[0]);
+    } else {
+      setIsError(true);
+      notifications.toasts.addError(new Error('Fleet server host not found'), {
+        title: i18n.translate(
+          'xpack.fleet.agentEnrollment.cloudFormation.errorLoadingFleetServerHosts',
+          {
+            defaultMessage: 'Error while fetching Fleet Server hosts',
+          }
+        ),
+      });
+    }
+  }, [data?.item.fleet_server_hosts, isLoading, notifications.toasts]);
+
+  const cloudFormationUrl =
+    enrollmentAPIKey && fleetServer
+      ? createCloudFormationUrl(
+          cloudFormationTemplateUrl,
+          enrollmentAPIKey,
+          fleetServer,
+          kibanaVersion
+        )
+      : '';
+
+  return (
```

Comment:
Since the rendering of this component depends on `cloudFormationUrl` above, which in turn depends on `fleetServer` which is fetched from the network, we need to render a loading state here. e.g.

```tsx
if (isLoading) {
  // Or some other `EuiSkeleton` components that make sense with the content
  return <EuiSkeletonText lines={4} /> 
}

return (
  <>
    <EuiSpacer size="m" />
    ...
  </>
)
```

Without this, the first render here is going to spit out a url with `undefined` in it for the Fleet Server. It's unlikely it would be clicked by a user, but let's prevent that invalid URL from appearing in the UI at all.